src/main.rs

```diff
@@ -284,17 +293,32 @@ fn get_includes_from_linker_script(linker_script: &str) -> Vec<&str> {
 /// Looks for "RAM : ORIGIN = $origin, LENGTH = $length"
 // FIXME this is a dumb line-by-line parser
 fn find_ram_in_linker_script(linker_script: &str) -> Option<MemoryEntry> {
-    macro_rules! tryc {
-        ($expr:expr) => {
-            if let Ok(x) = $expr {
-                x
-            } else {
-                continue;
-            }
-        };
+    let mut index_line: usize = 0;
+    for (i, line) in linker_script.lines().enumerate() {
+        if let Some(_pos) = line.find("RAM") {
+            index_line = i;
+        }
     }
 
-    for (index, mut line) in linker_script.lines().enumerate() {
+    // Removing all white spaces of the file and replacing with a space
+    let re = Regex::new(r"[^\S]+").unwrap();
+    let linker_script = &re.replace_all(linker_script, " ").to_string();
```

Comment:
By collapsing everything into a single line, the returned line will always be `0`, and the code replacing that line in the source linker script will replace the wrong line.

Reply:
> I fail to remove comments properly. Tried with Regex but I run into ownership issues.

The "proper" way would be to have a lexer in addition to the parser and remove the comments at the lexer level, but that might add too much complexity for this (could a knurling maintainer chime in on that?).

If the decision is to just have the recursive descent parser without a lexer in front of it, a way to handle that would be to have `eat_whitespace` also remove comments, and to have a custom `find()` that skips the sections of the input inside a comment.

Reply:
Ok.
What about the other problem?

Aka the `struct Parser` that provokes other errors?

Reply:
> > I fail to remove comments properly. Tried with Regex but I run into ownership issues.
> 
> The "proper" way would be to have a lexer in addition to the parser and remove the comments at the lexer level, but that might add too much complexity for this (could a knurling maintainer chime in on that?).
> 
> If the decision is to just have the recursive descent parser without a lexer in front of it, a way to handle that would be to have `eat_whitespace` also remove comments, and to have a custom `find()` that skips the sections of the input inside a comment.

Hello!
Can unfortunately not transfer ownership:
https://github.com/Dajamante/linker_flip_link